articles/client-platforms/angular2/05-calling-apis.md

```diff
@@ -0,0 +1,125 @@
+
+#### Meet AuthHttp
+
+In order to make an authorized request, [angular2-gwt](https://github.com/auth0/angular2-jwt) provides the `AuthHttp` helper which automatically add the authorization header to the requests.
+Making secure HTTP calls looks the same as it would with regular Http.
+
+First you need to add the `AUTH_PROVIDERS` from `angular-gwt`
+
+${snippet(meta.snippets.auth-providers-setup)}
+
+
+Then you can import `AuthHttp` in your component and make the authenticated request:
+
+
+```typescript
+import {Component} from '@angular/core';
+import {Auth} from './auth.service';
+import {AuthHttp} from 'angular2-jwt';
+import 'rxjs/add/operator/map';
+
+@Component({
+  selector: 'ping',
+  templateUrl: 'app/ping.template.html'
+})
+export class Ping {
+  API_URL: string = 'http://localhost:3001';
+  message: string;
+
+  constructor(private auth: Auth, private authHttp: AuthHttp) {}
+
+  public securedPing() {
+    this.authHttp.get(`${this.API_URL}/secured/ping`)
+      .map(res => res.json())
+      .subscribe(
+        data => this.message= data.text,
+        error => this.message = error._body || error
+      );
+  }
+}
+```
+
+With that your request will have Authorization in your request herders:
+
+```bash
+Authorization: Bearer eyJ0eXAiOiJKV1Qi...
+```
+
+By default `AuthHttp` will fetch the token from `localStorage`, using `'id_token'` key. You can change the key used, or even set another function to get the token, setting the provider by yourself. More detailed options [here](https://github.com/auth0/angular2-jwt#configuration-options)
+
+
+For example changing the tokenName (key used in localStorage):
+
+
+```typescript
+<!-- ===== ./auth.service.ts ===== -->
+...
+localStorage.setItem('another_token_name', authResult.idToken);
+...
+
+<!-- ===== ./main.ts ===== -->
+bootstrap(AppComponent, [
+  ...
+  provide(AuthHttp, {
+    useFactory: (http) => {
+      return new AuthHttp(new AuthConfig({
+        tokenName: 'another_token_name'
+      }), http);
+    },
+    deps: [Http]
+  }),
+  ...
+])
+```
+
+#### Not sending the JWT for specific requests
+
+When you don't want to send the authentication headers just use the default angular [Http](https://angular.io/docs/ts/latest/guide/server-communication.html) client.
+
+```typescript
+import {Component} from '@angular/core';
+import {Auth} from './auth.service';
+import {AuthHttp} from 'angular2-jwt';
+import {Http} from '@angular/http';
+import 'rxjs/add/operator/map';
+
+@Component({
+  selector: 'ping',
+  templateUrl: 'app/ping.template.html'
+})
+export class Ping {
+  API_URL: string = 'http://localhost:3001';
+  message: string;
+
+  constructor(private auth: Auth, private http: Http, private authHttp: AuthHttp) {}
+
+  public ping() {
+    var url = 'http://localhost:3001/ping';
+    this.http.get(`${this.API_URL}/ping`)
```

Comment:
don't you think it would be nice to have some code comments informing the differences?